state_handler/state_handler.go

```diff
@@ -85,6 +86,14 @@ func (sh DefaultStateHandler) Generate(cookieStore *sessions.CookieStore, w http
 		continueURI = r.RequestURI
 	}
 
+	// retrieve redirect url from query in order to let frontend decide where to redirect
+	q := r.URL.Query()
+	redirectUrl := q.Get(continueParam)
```

Comment:
I think use name `continueArg` would be better

Reply:
done

Reply:
done e78132c6374b8ba33b680e5e4850f2ec0cc773f4